src/dispatch/signal/service.py

```diff
@@ -371,9 +372,9 @@ def create(*, db_session: Session, signal_in: SignalCreate) -> Signal:
     signal.entity_types = entity_types
 
     engagements = []
-    for eng in signal_in.engagements:
+    for signal_engagement_in in signal_in.engagements:
         signal_engagement = get_signal_engagement_by_name(
-            db_session=db_session, project_id=project.id, signal_engagement_in=eng
+            db_session=db_session, project_id=project.id, name=signal_engagement_in.name
```

Comment:
Is this the bug?

Reply:
I'm not 100% sure as this is inside the create function, but I didn't find any issues with the update one.

Reply:
I think I found the issue in update. We're actually not updating the engagement if we find it: https://github.com/Netflix/dispatch/blob/master/src/dispatch/signal/service.py#L456

Reply:
We should update the engagement here if we found it https://github.com/Netflix/dispatch/blob/master/src/dispatch/signal/service.py#L461

Reply:
I think we have the same issue with entity types, filters, and workflows...

Reply:
We don't update them before appending...

Reply:
I'll merge this PR and create another one on Monday.

Reply:
Sounds good, thanks for tracking it down